subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -939,7 +939,7 @@ int lwm2m_init_message(struct lwm2m_message *msg)
 	 * msg->tkl == 0 is for a new TOKEN
 	 * msg->tkl == LWM2M_MSG_TOKEN_LEN_SKIP means dont set
 	 */
-	if (msg->tkl == 0U) {
+	if (msg->tkl == LWM2M_MSG_TOKEN_GENERATE_NEW) {
 		tokenlen = 0U;
```

Comment:
Looking at this line, I already see that token handling in the lwm2m library also needs an inspection. On one hand, we generate a token, on the other `tokenlen` is set to 0. As a result, we don't send a token anyway... (just confirmed in wireshark)

Reply:
Yes, I didn't mean to have this fixed in this PR. I'll create a separate issue for token handling improvement.